test/stream_test.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:test/test.dart';
+import 'dart:async';
+import 'entity.dart';
+import 'test_env.dart';
+import 'objectbox.g.dart';
+import 'package:objectbox/src/observable.dart';
+
+// ignore_for_file: non_constant_identifier_names
+
+void main() {
+  TestEnv env;
+  Box box;
+
+  setUp(() {
+    env = TestEnv('streams');
+    box = env.box;
+  });
+
+  test('Subscribe to stream of entities', () async {
+
+    final result = <String>[];
+    final text = TestEntity_.tString;
+    final condition = text.notNull();
+    final query = box.query(condition).order(text).build();
+    final queryStream = query.findStream();
+    final subscription = queryStream.listen((list) {
+      final str = list.map((t) => t.tString).toList().join(', ');
+      result.add(str);
+    });
+
+    box.put(TestEntity(tString: 'Hello world'));
+    box.putMany(<TestEntity>[ TestEntity(tString: 'Goodbye'),
+      TestEntity(tString: 'for now') ]);
+
+    await Future.delayed(Duration(seconds: 0)); // ffi explodes without
+    expect(result,
+        ['for now, Goodbye, Hello world', 'for now, Goodbye, Hello world']);
+
+    await subscription.cancel();
+  });
+
+  test('Subscribe to stream of query', () async {
+
+    final result = <int>[];
+    final text = TestEntity_.tString;
+    final condition = text.notNull();
+    final query = box.query(condition).order(text).build();
+    final queryStream = query.stream;
+    final subscription = queryStream.listen((query) {
+      result.add(query.count());
+    });
+
+    box.put(TestEntity(tString: 'Hello world'));
+
+    // idem, see above
+    box.putMany(<TestEntity>[ TestEntity(tString: 'Goodbye'),
+      TestEntity(tString: 'for now') ]);
+
+    await Future.delayed(Duration(seconds: 0)); // ffi explodes without
+    expect(result, [3, 3]);
```

Comment:
Same here, shouldn't the counts be `[1, 3]`?

Reply:
Doh again.

Reply:
I don't know whether this is a dart test async or OB sync issue. I tried setting it up on the flutter demo app, after updating dart/flutter, [but android needs updating again](https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects).

Also there's a non-nullness feature requirement that's blocking me.

Dart SDK version: 2.9.3 (stable) (Tue Sep 8 11:21:00 2020 +0200) on "macos_x64"
Flutter 1.22.0-10.0.pre.384 • channel master • https://github.com/flutter/flutter.git
Framework • revision 062e469f1b (15 hours ago) • 2020-09-27 15:32:07 -0700
Engine • revision 3a73d073c8
Tools • Dart 2.11.0 (build 2.11.0-161.0.dev)

I setup a new flutter android/iOS app and I'll try again.

Reply:
You're right; I fixed it with a delay by 0.